docs/DeveloperGuide.md

```diff
@@ -1,38 +1,254 @@
+is "deposit." Based on the second word, a method in `DepositAction` class is called corresponding to the command
+requested by the user.
 
-|Version| As a ... | I want to ... | So that I can ...|
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+Attached below is how the `DepositCommand` class is implemented along with its relation with the other `Data` classes
+and the abstract
+`Command` class.
 
-## Non-Functional Requirements
+![DepositCommand.png](uml/DepositCommand.png)
 
-{Give non-functional requirements}
+### Design & Implementation of the Deposit Feature
 
-## Glossary
+Like all other functionalities of BudgetBuddy, the deposit feature is heavily modularized and designed with an OOP lens.
+Because of this,
+the parsing of user input, the parsers for each feature, and each feature's actions are all in separate classes.
 
-* *glossary item* - Definition
+On a high level, the deposit feature starts with `CommandParser` taking in the input and choosing which `Command` class
+to execute from.
+This happens with all user input in BudgetBuddy. Then if the first word is "deposit," the `execute` function of
+the `DepositCommand` class
+will run, creating a new `DepositAction` class. The `execute` function will run a method corresponding to what the user
+inputs. The design
+of this three class system is meant to modularize the different aspects of the internal logic so future problems would
+be encapsulated in a specific location.
 
-## Instructions for manual testing
+### Expense Component
+
+### ExpenseCommand Class
+
+The `ExpenseCommand` class contains methods that relate to the execution of the expense functionality in BudgetBuddy.
+Users can
+create new expense entries with a timestamp (if necessary). These expenses are then stored in an expense list and users
+can delete existing
+expenses, find expenses using keywords, and list all expenses according to their categories.
+
+The user input is split by the parser in the `CommandParser` class and is redirected to the `ExpenseCommand` class if
+the first word is
+"expense". Based on the second command word, a method in `CommandAction` class is called corresponding to the command
+entered by the user.
+
+Attached below is how the `ExpenseCommand` class is implemented along with its relation with the other `Data` classes
+and the abstract
+`Command` class.
+
+![ExpenseCommand.png](uml/ExpenseCommand.png)
+
+### ExpenseCommand Sequence
+
+Once a string `input` from the user has been deemed as an `Expense` command, the `input` will be passed into the
+`ExpenseCommand` class. This class determines what expense commands to carry out to update the `data` stored in the
+`Budget` and `Expense` array lists, as shown in the sequence diagram below.
+
+![img.png](uml/ExpenseCommandSequence.png)
```

Comment:
this diagram has too much going and it is very difficult to comprehend the command sequence, I would suggest breaking down the flow into smaller sub portions and explaining how it works using cropped images of the diagram.